File: parse_requirements.py
import openpyxl
import json

def parse_excel(file_path):
    """解析Excel文件并输出内容"""
    try:
        # 加载工作簿
        workbook = openpyxl.load_workbook(file_path)
        
        result = {
            "工作表列表": workbook.sheetnames,
            "数据": {}
        }
        
        # 遍历所有工作表
        for sheet_name in workbook.sheetnames:
            sheet = workbook[sheet_name]
            print(f"\n{'='*50}")
            print(f"工作表: {sheet_name}")
            print(f"{'='*50}")
            
            sheet_data = []
            
            # 读取所有行
            for row_idx, row in enumerate(sheet.iter_rows(values_only=True), start=1):
                # 过滤空行
                if any(cell is not None for cell in row):
                    print(f"第{row_idx}行: {row}")
                    sheet_data.append(row)
            
            result["数据"][sheet_name] = sheet_data
        
        # 保存为JSON文件
        with open('需求解析结果.json', 'w', encoding='utf-8') as f:
            json.dump(result, f, ensure_ascii=False, indent=2, default=str)
        
        print(f"\n{'='*50}")
        print("解析完成！结果已保存到 '需求解析结果.json'")
        print(f"{'='*50}")
        
        return result
        
    except Exception as e:
        print(f"解析出错: {str(e)}")
        return None

if __name__ == "__main__":
    parse_excel("需求.xlsx")

File: parse_requirements.js
const XLSX = require('xlsx');
const fs = require('fs');

function parseExcel(filePath) {
    try {
        console.log('正在读取Excel文件...');
        
        // 读取Excel文件
        const workbook = XLSX.readFile(filePath, { cellDates: true });
        
        const result = {
            工作表列表: workbook.SheetNames,
            数据: {}
        };
        
        // 遍历所有工作表
        workbook.SheetNames.forEach(sheetName => {
            console.log(`\n${'='.repeat(50)}`);
            console.log(`工作表: ${sheetName}`);
            console.log('='.repeat(50));
            
            const sheet = workbook.Sheets[sheetName];
            
            // 转换为JSON格式
            const jsonData = XLSX.utils.sheet_to_json(sheet, { 
                header: 1,
                defval: null,
                raw: false
            });
            
            // 打印数据
            jsonData.forEach((row, idx) => {
                if (row.some(cell => cell !== null && cell !== '')) {
                    console.log(`第${idx + 1}行:`, row);
                }
            });
            
            result.数据[sheetName] = jsonData;
        });
        
        // 保存为JSON文件
        fs.writeFileSync(
            '需求解析结果.json', 
            JSON.stringify(result, null, 2),
            'utf-8'
        );
        
        console.log(`\n${'='.repeat(50)}`);
        console.log("解析完成！结果已保存到 '需求解析结果.json'");
        console.log('='.repeat(50));
        
        return result;
        
    } catch (error) {
        console.error('解析出错:', error.message);
        return null;
    }
}

// 执行解析
parseExcel('需求.xlsx');
